server/src-lib/Hasura/RQL/Types/Common.hs

```diff
@@ -277,12 +278,26 @@ isSystemDefined = unSystemDefined
 successMsg :: EncJSON
 successMsg = "{\"message\":\"success\"}"
 
+newtype NonNegativeInt = NonNegativeInt { getNonNegativeInt :: Int }
```

Comment:
There should be a function `mkNonNegativeInt` which takes an `Int` and then returns a `Maybe Int`. The function should return `Nothing` when a negative value is passed otherwise it should return `Just x`. Check out `mkNonEmptyText` for reference.

Reply:
Just a small question, `mkNonNegativeInt` should return `Maybe NonNegativeInt` right? Instead of `Maybe Int`?

Reply:
Ah sorry, Yes, you're right, it should return `Maybe NonNegativeInt`